grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,567 @@
+        assert ma_env._subgrids_cls['action']['agent_0'].n_gen == 3
+        assert ma_env._subgrids_cls['action']['agent_1'].n_gen == 3
+        
+        self.checks(ma_env, action_domains)
+        
+        # We compare the interconnection original line ids with known values for every agent
+        assert (ma_env._subgrids_cls['action']['agent_0'].interco_to_lineid == np.array([15,16,17])).all()
+        assert (ma_env._subgrids_cls['action']['agent_1'].interco_to_lineid == np.array([15,16,17])).all()
+        
+        ref = np.array([[ 0., -1., -1.,  0., -1., -1, -1.],
+                        [ 0., -1., -1.,  1., -1., -1, -1.],
+                        [ 0., -1.,  2., -1., -1., -1, -1.],
+                        [ 1., -1., -1., -1.,  0., -1, -1.],
+                        [ 1., -1., -1.,  2., -1., -1, -1.],
+                        [ 1., -1., -1.,  3., -1., -1, -1.],
+                        [ 1., -1., -1.,  4., -1., -1, -1.],
+                        [ 1., -1.,  0., -1., -1., -1, -1.],
+                        [ 1.,  0., -1., -1., -1., -1, -1.],
+                        [ 2., -1., -1., -1.,  2., -1, -1.],
+                        [ 2., -1., -1.,  5., -1., -1, -1.],
+                        [ 2., -1.,  1., -1., -1., -1, -1.],
+                        [ 2.,  1., -1., -1., -1., -1, -1.],
+                        [ 3., -1., -1., -1.,  3., -1, -1.],
+                        [ 3., -1., -1., -1.,  5., -1, -1.],
+                        [ 3., -1., -1.,  6., -1., -1, -1.],
+                        [ 3., -1., -1., -1., -1., -1,  0.],
+                        [ 3., -1., -1., -1., -1., -1,  1.],
+                        [ 3.,  2., -1., -1., -1., -1, -1.],
+                        [ 4., -1., -1., -1.,  1., -1, -1.],
+                        [ 4., -1., -1., -1.,  4., -1, -1.],
+                        [ 4., -1., -1., -1.,  6., -1, -1.],
+                        [ 4., -1., -1., -1., -1., -1,  2.],
+                        [ 4.,  3., -1., -1., -1., -1, -1.]])
+        # We compare with a known value
+        assert (ma_env._subgrids_cls['action']['agent_0'].grid_objects_types == ref).all()
+        
+        
+    def test_build_subgrid_obj2(self):    
+        # 2
+        # Test with 3 agents
+        action_domains = {
+            'test_2_agent_0' : [0,1, 2, 3, 4],
+            'test_2_agent_1' : [5, 6, 7, 8, 9],
+            'test_2_agent_2' : [10, 11, 12, 13],
+            
+        }
+        ma_env = MultiAgentEnv(self.env, action_domains, _add_to_name="test_build_subgrid_obj2")
+        
+        self.checks(ma_env, action_domains)
+
+    
+    def test_build_subgrid_obj3(self):    
+        # 3 random sub ids
+        np.random.seed(0)
+        for it in range(10):
+            sub_ids = list(range(14))
+            np.random.shuffle(sub_ids)  # you should see it for reproducible results
+            action_domains = {
+                'agent_0' : sub_ids[:7],
+                'agent_1' : sub_ids[7:],
+            }
```

Comment:
Can you "increase randomness" and make the "7" here "random" too (so draw an integer between 1 and 12 (included) at random instead of hard coding "7"

Reply:
Ok done

Reply:
Done